### ISSUE TITLE
Add theming support

### ISSUE THREAD

**Issue by paddymul** [buckaroo-data/buckaroo#238]
### Checks

- [X] I have checked that this enhancement has not already been requested

### How would you categorize this request.  You can select multiple if not sure

Display (is this related to visual display of a value)

### Enhancement Description

Buckaroo has a default dark-mode look.  This works for my uses.  It would be good to add theming and styling support.

This could probably go in two phases, first - light/dark switch.  Then more comprehensive theming from the python side.

### Pseudo Code Implementation

Theming probably falls into another baselevel key for `dfviewer_config`

```typescript
export type GenericViewerConfig {
  theme: "dark" | "light";
}

export interface DFViewerConfig {
  pinned_rows: PinnedRowConfig[];
  column_config: ColumnConfig[];
  extra_grid_config?: GridOptions;
  generic_config?: GenericViewerConfig
}
```

Later it could look like this 
```typescript
export type GenericViewerConfig {
  theme: "dark" | "light";
  base_css_classname?: string;
  extra_css?: string;
  extra_css_url?: string

}

```

### Prior Art

Not so much prior art but implementation plan.

This is the type of task that someone who knows more about styling and cares more about it should implement.

**Comment by bayesianbias** [comment]
Hello!
I find it very useful to be able to change the default theme to a light one to integrate it more easily in contexts with light themes. 

While this option is being implemented, is there a simple way to change the default theme from ag-theme-alpine-dark to ag-theme-alpine? I’ve seen that it’s a matter of changing the class in the container div, but I’m using Solara with SolaraDFViewer, and I see that the widget always applies the dark theme, and I can’t find a way to change it. Do I need to inject JavaScript to change it?

Javascript is not my thing so sorry for this basic questions...  would something along these lines work? but when does it need to be applied before the page renders to avoid color "flashes"?

```
const gridElement = document.querySelector('.ag-theme-alpine, .ag-theme-balham');

// Remove existing theme classes
gridElement.classList.remove('ag-theme-alpine', 'ag-theme-alpine-dark');

// Add the new theme class
gridElement.classList.add(newTheme);

// Refresh grid size to reflect theme changes
if (gridApi) {
    gridApi.setDomLayout();
}
```

Any ideas on to change it specifically for this context?
Thanks!

**Comment by paddymul** [comment]
Sorry, I just saw this feature request.

I think I could include it as an extra option to DFViewer.  

The relevant JS code is here
https://github.com/paddymul/buckaroo/blob/main/js/components/DFViewerParts/DFViewer.tsx#L133-L137
and
https://github.com/paddymul/buckaroo/blob/main/js/components/DFViewerParts/DFViewer.tsx#L51-L57

Both classes need to be changed together to have the hilighting work properly ( I think)

I could add them as specific options in `extra_grid_config`.  but I'm a bit nervous about that,  I don't want that key to expand enormously, and I'm worried about conflict with built in ag-grid config names.

What would you want the interface to look like from `SolaraDFViewer`?

**Comment by paddymul** [comment]
This will be coming out in a release soon.  You will have to handle getting CSS onto the page.

```
BuckarooWidget(obj_df, 
               component_config={'className':'asdf', 
                                 'selectionBackground': 'red'})
```
<img width="624" alt="Screenshot 2024-10-31 at 4 36 00 PM" src="https://github.com/user-attachments/assets/c7e4678e-0b7c-4e33-8cc5-d05a8cd4d0f6">

**Comment by paddymul** [comment]
Fixed by da1bf14 release 0.7.9
https://github.com/paddymul/buckaroo/commit/da1bf14d5b64660eb9a314aadefffd597c04b737
